pkg/resources/health.go

```diff
@@ -37,6 +38,10 @@ func HealthyDeployment(ctx context.Context, client ctrlruntimeclient.Client, nn
 		return kubermaticv1.HealthStatusDown, err
 	}
 
+	if minReady < 0 {
+		minReady = *deployment.Spec.Replicas
+	}
+
```

Comment:
We have like 3 or 4 different functions to check if a deployment is "healthy". I chose to use this and extend it a bit to let it auto-discover the intended number of replicas. Lots of Go stdlib functions also use "-1" for "all", so I didn't feel too bad to introduce this special case.